File: apps/api/app/models/enums.py
import enum


class UserRole(str, enum.Enum):
    ADMIN = "admin"
    THERAPIST = "therapist"
    PATIENT = "patient"

class AppointmentStatus(str, enum.Enum):
    SCHEDULED = "scheduled"
    CONFIRMED = "confirmed"
    IN_PROGRESS = "in_progress"
    COMPLETED = "completed"
    CANCELLED = "cancelled"
    MISSED = "missed"

class AssessmentType(str, enum.Enum):
    INITIAL = "initial"
    WEEKLY = "weekly"
    FINAL = "final"

class TreatmentStatus(str, enum.Enum):
    PLANNED = "planned"
    IN_PROGRESS = "in_progress"
    COMPLETED = "completed"
    CANCELLED = "cancelled"

class NotificationType(str, enum.Enum):
    EMAIL = "email"
    PUSH = "push"
    IN_APP = "in_app"

class NotificationStatus(str, enum.Enum):
    UNREAD = "unread"
    READ = "read"
    ARCHIVED = "archived"

class ConsentStatus(str, enum.Enum):
    PENDING = "pending"
    SIGNED = "signed"
    REVOKED = "revoked"

class SessionStatus(str, enum.Enum):
    PLANNED = "planned"
    COMPLETED = "completed"
    CANCELLED = "cancelled"
    NO_SHOW = "no_show"

class Gender(str, enum.Enum):
    MALE = "male"
    FEMALE = "female"
    OTHER = "other"
    PREFER_NOT_TO_SAY = "prefer_not_to_say"

class BloodGroup(str, enum.Enum):
    A_POS = "A+"
    A_NEG = "A-"
    B_POS = "B+"
    B_NEG = "B-"
    AB_POS = "AB+"
    AB_NEG = "AB-"
    O_POS = "O+"
    O_NEG = "O-"
    UNKNOWN = "unknown"

class DocumentType(str, enum.Enum):
    MRI = "mri"
    X_RAY = "x_ray"
    PRESCRIPTION = "prescription"
    LAB_REPORT = "lab_report"
    INSURANCE = "insurance"
    REFERRAL = "referral"
    OTHER = "other"

class MediaType(str, enum.Enum):
    IMAGE = "image"
    VIDEO = "video"
    AUDIO = "audio"

class TherapistStatus(str, enum.Enum):
    ACTIVE = "active"
    INACTIVE = "inactive"
    ON_LEAVE = "on_leave"

class InvoiceStatus(str, enum.Enum):
    DRAFT = "draft"
    ISSUED = "issued"
    PARTIALLY_PAID = "partially_paid"
    PAID = "paid"
    OVERDUE = "overdue"
    CANCELLED = "cancelled"

class PaymentStatus(str, enum.Enum):
    PENDING = "pending"
    COMPLETED = "completed"
    FAILED = "failed"
    REFUNDED = "refunded"

class PaymentMethod(str, enum.Enum):
    CASH = "cash"
    CARD = "card"
    UPI = "upi"
    BANK_TRANSFER = "bank_transfer"

class DevicePlatform(str, enum.Enum):
    ANDROID = "android"
    IOS = "ios"
    WEB = "web"

class OtpPurpose(str, enum.Enum):
    LOGIN = "login"
    REGISTRATION = "registration"
    FORGOT_PASSWORD = "forgot_password"

class StrengthScale(str, enum.Enum):
    ZERO = "0"
    TRACE = "1"
    POOR = "2"
    FAIR = "3"
    GOOD = "4"
    NORMAL = "5"

File: packages/shared-types/src/enums.ts
export const UserRoles = ['admin', 'therapist', 'patient'] as const;
export type UserRole = (typeof UserRoles)[number];

export const AppointmentStatuses = [
  'scheduled',
  'confirmed',
  'in_progress',
  'completed',
  'cancelled',
  'missed',
] as const;
export type AppointmentStatus = (typeof AppointmentStatuses)[number];

export const AssessmentTypes = ['initial', 'weekly', 'final'] as const;
export type AssessmentType = (typeof AssessmentTypes)[number];

export const TreatmentStatuses = ['planned', 'in_progress', 'completed', 'cancelled'] as const;
export type TreatmentStatus = (typeof TreatmentStatuses)[number];

export const NotificationTypes = ['email', 'push', 'in_app'] as const;
export type NotificationType = (typeof NotificationTypes)[number];

export const NotificationStatuses = ['unread', 'read', 'archived'] as const;
export type NotificationStatus = (typeof NotificationStatuses)[number];

export const ConsentStatuses = ['pending', 'signed', 'revoked'] as const;
export type ConsentStatus = (typeof ConsentStatuses)[number];

export const SessionStatuses = ['planned', 'completed', 'cancelled', 'no_show'] as const;
export type SessionStatus = (typeof SessionStatuses)[number];

export const Genders = ['male', 'female', 'other', 'prefer_not_to_say'] as const;
export type Gender = (typeof Genders)[number];

export const BloodGroups = ['A+', 'A-', 'B+', 'B-', 'AB+', 'AB-', 'O+', 'O-', 'unknown'] as const;
export type BloodGroup = (typeof BloodGroups)[number];

export const DocumentTypes = [
  'mri',
  'x_ray',
  'prescription',
  'lab_report',
  'insurance',
  'referral',
  'other',
] as const;
export type DocumentType = (typeof DocumentTypes)[number];

export const MediaTypes = ['image', 'video', 'audio'] as const;
export type MediaType = (typeof MediaTypes)[number];

export const TherapistStatuses = ['active', 'inactive', 'on_leave'] as const;
export type TherapistStatus = (typeof TherapistStatuses)[number];

export const InvoiceStatuses = [
  'draft',
  'issued',
  'partially_paid',
  'paid',
  'overdue',
  'cancelled',
] as const;
export type InvoiceStatus = (typeof InvoiceStatuses)[number];

export const PaymentStatuses = ['pending', 'completed', 'failed', 'refunded'] as const;
export type PaymentStatus = (typeof PaymentStatuses)[number];

export const PaymentMethods = ['cash', 'card', 'upi', 'bank_transfer'] as const;
export type PaymentMethod = (typeof PaymentMethods)[number];

export const DevicePlatforms = ['android', 'ios', 'web'] as const;
export type DevicePlatform = (typeof DevicePlatforms)[number];

export const OtpPurposes = ['login', 'registration', 'forgot_password'] as const;
export type OtpPurpose = (typeof OtpPurposes)[number];

export const StrengthScales = ['0', '1', '2', '3', '4', '5'] as const;
export type StrengthScale = (typeof StrengthScales)[number];
